[BUGFIX] Add demo_cp_ct_surface location to data files.

misi9170 opened the pull request:
I believe this should address #1094 . I'm not sure what the best way to test the fix is, though, short of merging this and creating a new release (UPDATE: Read comments below for how to test locally).

UPDATE:
To avoid this type of bug being introduced again, as well as adding the missing data to the `[tool.setuptools.package-data]` list of pyproject.toml, this PR:
- Makes github workflow installations of floris non-editable by removing `-e` flag
- Moves tests into a subdirectory to avoid finding the data (still don't totally understand why this is necessary, but it has the desired effect).

rafmudaf commented:
You can test this locally by doing `pip install floris/` without the typical `-e` in a new Python environment. Then, you can search for the file in the installation directory to verify it's there and also run the demo case.

misi9170 commented:
@rafmudaf 's testing suggestion did work locally. I'm now testing removing the `-e` flag from github workflows to see if that would have identified this issue earlier (thanks, @rafmudaf , for the suggestion). Assuming the tests fail in the expected way, I'll re-add the package data. If test/examples then pass, this should be good to go.

misi9170 commented:
Expected failures are now happening. Now re-adding the data.

misi9170 commented:
Ok, after a lot of digging, @rafmudaf and I found and a couple of other issues:
- added wrg_test.wrg to tests/data to avoid relying on data in the examples directory
- added `-rL` flag to copies to resolve symbolic links
- add turbine_library/*.csv to `[tool.setuptools.package-data]` so that multimensional turbine data is available with Floris is installed using PyPI
- created more robust paths using the package root for data loading in tests
